listener/listener.go

```diff
@@ -234,19 +243,36 @@ func (l *Listener) Listen(ctx context.Context, handler message.Handler, topicNam
 		return fmt.Errorf("failed to create new subscription receiver %s: %w", l.subscriptionEntity.Name, err)
 	}
 
-	// Create a handle class that has that function
-	listenerHandle := subReceiver.Listen(ctx, servicebus.HandlerFunc(
-		func(ctx context.Context, msg *servicebus.Message) error {
-			currentHandler := handler
-			for !message.IsDone(currentHandler) {
-				currentHandler = currentHandler.Do(ctx, handler, msg)
-				// handle nil as a Completion!
-				if currentHandler == nil {
-					currentHandler = message.Complete()
+	//blantantly stolen from https://github.com/Azure/azure-service-bus-go/pull/117/files
+	msgChan := make(chan *servicebus.Message, l.concurrency)
+	// Define a function that should be executed when a message is received.
+	var concurrentHandler servicebus.HandlerFunc = func(ctx context.Context, msg *servicebus.Message) error {
+		msgChan <- msg
+		return nil
+	}
+	if l.concurrency == 0 {
+		l.concurrency = 5 //what's the right default? 1? something higher?
```

Comment:
this actually needs to be pretty high or retry later will still sink us.

Reply:
We could remove any max concurrancy here and allways spin up as many go functions as we have messages. That is closer to what charlie was considering.